packages/preservation_spine/tests/test_checkpoint.py

```
@@ -0,0 +1,733 @@
+    calls = {"count": 0}
+    original_fsync = checkpoint_module._fsync_descriptor
+
+    def failing_fsync(fd: int) -> None:
+        calls["count"] += 1
+        if calls["count"] >= 2:
+            raise OSError("injected fsync failure")
+        original_fsync(fd)
+
+    monkeypatch.setattr(checkpoint_module, "_fsync_descriptor", failing_fsync)
+
+    with pytest.raises(CheckpointIntegrityError, match="fsync|append|write"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)